lib/msf_autoload.rb

```diff
@@ -0,0 +1,167 @@
+require "zeitwerk"
+
+class TempInflector < Zeitwerk::Inflector
+  def camelize(basename, abspath)
+    if basename == "osx" && abspath.end_with?("lib/msf/core/payload/osx", "lib/msf/core/payload/osx.rb")
+      "Osx"
+    elsif basename == "exe" && abspath.end_with?("lib/msf/core/exe", "lib/msf/core/exe.rb")
+      "Exe"
+    else
+      super
+    end
+  end
+end
+
+loader = Zeitwerk::Loader.new
+loader.push_dir("#{__dir__}/msf/core/", namespace: Msf)
+loader.push_dir("#{__dir__}/../app/validators/")
+
+loader.ignore("#{__dir__}/msf/core/constants.rb", "#{__dir__}/msf/core/cert_provider.rb", "#{__dir__}/msf/core/rpc/json/error.rb", "#{__dir__}/msf/core/rpc/json/v2_0/", "#{__dir__}/msf/core/modules/external/ruby/metasploit.rb", "#{__dir__}/msf/core/rpc/v10/constants.rb")
```

Comment:
If we split this on to new lines, it'll make future reviews easier when new items are added to this list :+1: